### PR TITLE
Fix * imports

### LINKED ISSUE #136: External modules
The code currently as `utils.modules_ext.py` to import external modules such as Numpy, Scipy, Matplotlib, etc. This means that modules are sometimes imported when not required, and also makes code linting very tricky. 

Instead, modules should be imported within the files in which they are used.

### PR DESCRIPTION
**Pull request by stefsmeets**
This PR fixes all the `from x import *` imports. Avoiding * imports is generally considered a [good practice](https://peps.python.org/pep-0008/#imports). I used [`ruff`](https://docs.astral.sh/ruff/rules/#isort-i) to help with the formatting, partly addressing #131 as well. 

To find import issues, you can use:

```
ruff check start_proteus.py  # single file
ruff check src/proteus       # directory
ruff check .                 # everything
```

Prepend `--fix` to sort imports and remove unused imports:

```
ruff check . --fix
```

I also set up a small [pre-commit](https://pre-commit.com/#usage) config so you can automatically have it sort imports on every commit, e.g.:

```
pre-commit install
```

Closes #136
Addresses #131

### Todo
- [x] Setup github action
- [x] Add documentation

### CODE REVIEW

**Comment by stefsmeets** [comment]
Hi @nichollsh @timlichtenberg This PR is ready, could you have a look and let me know what you think?

Could you test it with some of your workflows to make sure no imports were missed?

File: start_proteus.py [hunk 1]
```diff
@@ -3,21 +3,63 @@
...
+    ReadHelpfileFromCSV,
+    ReadInitFile,
+    SetDirectories,
+    UpdatePlots,
+    ValidateInitFile,
+    WriteHelpfileToCSV,
+    ZeroHelpfileRow,
+    parse_console_arguments,
+)
+from proteus.utils.logs import (
```
Comment (lsoucasse) [inline start_proteus.py:49]:
```suggestion
from proteus.utils.helper import (
    CleanDir,
    PrintHalfSeparator,
    PrintSeparator,
    safe_rm,
    UpdateStatusfile,
)
from proteus.utils.logs import (
```

**Review by lsoucasse — CHANGES_REQUESTED** [review]
Hi @stefsmeets, I spotted some missing imports. The default config script runs fine for me.

**Comment by stefsmeets** [comment]
Hi @lsoucasse thanks for pointing those out, I fixed the issues.

**Pull request merged**